File: src/main/java/bbarena/model/event/game/ChangeWeatherEvent.java
package bbarena.model.event.game;

import bbarena.model.Arena;
import bbarena.model.Weather;
import bbarena.model.util.Concat;
import bbarena.model.util.Pair;

public class ChangeWeatherEvent extends GameEvent {

    private Weather _newWeather;
    private Weather _oldWeather;

    public ChangeWeatherEvent() {
    }

    public void setNewWeather(Weather newWeather) {
        _newWeather = newWeather;
    }

    @Override
    protected void doEvent(Arena arena) {
        _oldWeather = arena.getWeather();
        arena.setWeather(_newWeather);
    }

    @Override
    protected void undoEvent(Arena arena) {
        arena.setWeather(_oldWeather);
    }

    @Override
    public String getString() {
        return Concat.buildLog(getClass(),
                new Pair("weather", _newWeather));
    }
}

File: src/main/kotlin/bbarena/model/event/game/ChangeWeatherEvent.kt
package bbarena.model.event.game

import bbarena.model.Arena
import bbarena.model.Weather
import bbarena.model.util.Concat

class ChangeWeatherEvent(var newWeather: Weather, var oldWeather: Weather) : GameEvent() {

    override val string: String
        get() = Concat.buildLog(javaClass,
                Pair("weather", newWeather))

    override fun doEvent(arena: Arena) {
        oldWeather = arena.weather
        arena.weather = newWeather
    }

    override fun undoEvent(arena: Arena) {
        arena.weather = oldWeather
    }
}
